### PR TITLE
Use right publish date format

### LINKED ISSUE #227: Wrong publication date in news sitemap
The _news:publication_date_ is set to UTC time, however, a wrong time offset is applied (see image below).

When using UTC, the time offset should be +00:00. Instead the time offset corresponding to the local time is outputted (+02:00) leading to a wrong publication time.

As a result, the article appears older in the search results, as if it had been published 2 hours before its actual publication time.

image

Related: https://github.com/Yoast/wpseo-news/issues/112

### PR DESCRIPTION
**Pull request by andizer**
## Summary

This PR can be summarized in the following changelog entry:
- [Bugfix] Fixes a bug where the wrong timezone has been used in the sitemap

## Relevant technical choices:

* Copied the logic from Yoast SEO to fallback by guessing the timezone

## Test instructions

This PR can be tested by following these steps:

* Add a post and see if the timezone will be correct.

Fixes #227

### CODE REVIEW

**Comment by CarolineGeven** [comment]
When using UTC + 2.00 in the settings, the time will be outputted correctly by WordPress. However, the sitemap itself still shows the incorrect date. 
schermafbeelding_2017-05-26_om_09_23_24

In the source, it says UTC is +1, while in fact it's +2.
From the source:
`2017-05-26T07:20:59+01:00`

**Comment by CarolineGeven** [comment]
@andizer I've checked the regular sitemap as we discussed above issue could be related to the sitemap in Yoast SEO for WordPress not setting the timezone correctly, but the timezone in the general sitemap works differently (it shows the correct time, but perhaps not in the correct way). My utc is set to +2, but my sitemap shows +1.

For example:
My post is posted at 10:12 hours utc +2. The news sitemap shows the time as 08:12 + 1, making it 09:12. The post sitemap shows the time as 09:12 +1, making the time 10:12, but not in the correct way.

I will open an issue in the WordPress SEO repo for this. When that is fixed, this one can be tested as well.

File: classes/class-sitemap.php [hunk 1]
```diff
@@ -329,17 +329,22 @@ private function wp_get_timezone_string() {
...
-		// Last try, guess timezone string manually.
-		if ( false === $timezone ) {
+		if ( false !== $timezone ) {
+			return $timezone;
+		}
 
-			if ( $timezone_id = $this->get_timezone_id( $utc_offset ) ) {
-				return $timezone_id;
+		// Last try, guess timezone string manually.
+		foreach ( timezone_abbreviations_list() as $abbr ) {
```
Comment (moorscode) [inline classes/class-sitemap.php]: Please revert and use the class method `get_timezone_id` for this.

File: classes/class-sitemap.php [hunk 2]
```diff
@@ -329,17 +329,22 @@ private function wp_get_timezone_string() {
 		}
 
 		// Adjust UTC offset from hours to seconds.
-		$utc_offset *= 3600;
+		$utc_offset *= HOUR_IN_SECONDS;
 
 		// @todo $timezone not being used when not false? JM
```
Comment (moorscode) [inline classes/class-sitemap.php]: Please remove comment

**Comment by IreneStr** [comment]
CR done 👍

**Comment by IreneStr** [comment]
CR :x:

### 1st post
Created at `09:13 UTC+0` with `UTC+0` in settings.

Source of post: ✅ 
schermafbeelding 2017-07-05 om 11 27 56

Sitemap time: ✅ 
schermafbeelding 2017-07-05 om 11 30 48


### 2nd post
Created at `09:18 UTC+0` with `UTC+2` in settings. 

Source of post: ❌ 
schermafbeelding 2017-07-05 om 11 29 21
Instead of 09:18+2, the source shows 11:18+0

Sitemap time: ❌ 
schermafbeelding 2017-07-05 om 11 31 45
Instead of 09:18+2, the sitemap shows 09:18+0

**Comment by atimmer** [comment]
The timezone of the site should be set to a city, not a UTC offset. See https://github.com/Yoast/wpseo-news/issues/306 to improve this. This can be tested again and if it works, be merged.

**Comment by moorscode** [comment]
Please fix the test (date expectation)

**Pull request merged**